File: l02/lista2.cpp
#include <iostream>
#include <vector>

std::vector<int> cutAndMend(std::vector<int> lst, int a, int b) {
    std::vector<int> result;
    for (int i = 0; i < lst.size(); i++) {
        if (i < a || i > b) {
            result.push_back(lst[i]);
        }
    }
    return result;
}

File: l02/lista2.clj
(define (cutAndMend lst a b)
    (define (aux lst current)
        (cond ((null? lst) '())
              ((and (>= current a) (<= current b)) (aux (cdr lst) (+ current 1)))
              (else (cons (car lst) (aux (cdr lst) (+ current 1))))))
    (aux lst 0))